File: Android applications/p0271getintentaction/java/com/korobeynikov/p0271getintentaction/info.java
package com.korobeynikov.p0271getintentaction;

import java.sql.Date;
import java.text.SimpleDateFormat;
import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class info extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.info);
        Intent intent=getIntent();
        String action=intent.getAction();
        String format="",textInfo="";
        if(action.equals("com.korobeynikov.intent.action.showTime")){
            format="HH:mm:ss";
            textInfo="Time: ";
        }
        else if(action.equals("com.korobeynikov.intent.action.showDate")){
            format="dd.MM.yyyy";
            textInfo="Date: ";
        }
        SimpleDateFormat sdf=new SimpleDateFormat(format);
        String dateTime=sdf.format(new Date(System.currentTimeMillis()));
        TextView tvDate=findViewById(R.id.tvInfo);
        tvDate.setText(textInfo+dateTime);
    }
}

File: KotlinLessons/AndroidLessons/p0271getintentaction/java/com/korobeynikov/p0271getintentaction/Info.kt
package com.korobeynikov.p0271getintentaction

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.info.*
import java.text.SimpleDateFormat
import java.util.*

class Info : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.info)
        val intent=getIntent()
        val action=intent.action
        var format=""
        var textInfo=""
        if(action.equals("com.korobeynikov.intent.action.showtime")){
            format="HH:mm:ss"
            textInfo="Time: "
        }
        else if(action.equals("com.korobeynikov.intent.action.showdate")){
            format="dd.MM.yyyy"
            textInfo="Date: "
        }
        val sdf=SimpleDateFormat(format)
        val datetime=sdf.format(Date(System.currentTimeMillis()))
        tvInfo.text=textInfo+datetime
    }
}